graphqlapi/src/main/java/io/stargate/graphql/web/resources/schemafirst/FilesResource.java

```diff
@@ -38,6 +50,27 @@ public Response getCqlDirectives() {
         .build();
   }
 
+  @GET
+  @Path("/namespace/{namespaceName}.graphql")
+  @Produces(MediaType.TEXT_PLAIN)
+  public Response getSchema(
```

Comment:
All of the endpoints in here should require the usual auth headers as well.

Reply:
👍 , this method should do the same check as `SchemaFetcher`.

`getCqlDirectives()` returns static contents that will be the same for every Stargate installation, so I think it  can skip auth.

Reply:
Are you sure that we can do the same check as `SchemaFetcher#authorize`?
I see a problem with it: we don't have the `AuthenticationSubject` in the `FilesResource`, but it's needed for the `authorize` check. Should we construct the `ImmutableAuthenticationSubject` somehow? Should the `FilesResource` create the `AuthenticationSubject` from the `X-Cassandra-Token` header?

2nd question: 
I don't see an `authZ` in any other `grapqhl-first` resources: (`NamespaceResource`, `AdminResource`).
Are we missing the `authZ` there as well?

Reply:
We do authentication and authorization in the REST API, it should be the same here.

Reply:
Is there an issue or TODO somewhere to also make sure this is added to `getCqlDirectives`?

Reply:
`getCqlDirectives()` serves generic content, from a file that is publicly available in this github repo.